lib/fetch/index.js

```diff
@@ -407,13 +400,6 @@ function fetching ({
   // 8. If request’s window is "client", then set request’s window to request’s
   // client, if request’s client’s global object is a Window object; otherwise
   // "no-window".
-  if (request.window === 'client') {
-    // TODO: What if request.client is null?
-    request.window =
-      request.client?.globalObject?.constructor?.name === 'Window'
-        ? request.client
-        : 'no-window'
-  }
```

Comment:
The problem with this kind of change is that it's hard to tell whether some other code indirectly inspects this and causes some externally observable changes... which is why IMHO it's better to just keep it and then we don't have to wonder.